src/Blackbird.py

```diff
@@ -16,12 +15,15 @@ class BlackBird(MCTS, Network):
     """ Class to train a network using an MCTS driver to improve decision making
     """
     class TrainingExample(object):
-        def __init__(self, state, value, childValues, probabilities, priors):
+        def __init__(self, state, value, childValues,
+                probabilities, priors, boardShape):
+            
             self.State = state
             self.Value = value
-            self.ChildValues = childValues.reshape((3,3)) if childValues is not None else None
+            self.BoardShape = boardShape
+            self.ChildValues = childValues.reshape(boardShape) if childValues is not None else None
             self.Reward = None
-            self.Priors = priors.reshape((3,3))
+            self.Priors = priors.reshape(boardShape)
```

Comment:
This comment applies to a couple places. Instead of defining a boardshape like this and reshaping it here, I think we should just offload the whole thing to the BoardState class.

This would involve doing a couple of things:
1. Create a new abstract class layer on top of GameState (or whatever thats called). There, we can put something like the following:
`def priorsToString(priors):
    return str(priors)
`
This gives a trivial default that is good enough if we don't feel like implementing it the future.
2. In both TicTacToe and Connect4, override it to reshape it 'properly'. 
This significantly reduces code coupling between the TrainingExample/Blackbird classes and the game definitions - keeping them strictly operational.

Reply:
I like this idea.  Thinking about it, it doesn't make sense for BlackBird to be doing any of the reshaping work; that should be taken care of in the appropriate BoardState handler.

Reply:
These changes have been made.  I'm not thrilled about the way it was done, and we'll need to refactor this once we do away with the `TrainingExample` class, in favor of something portable to server storage.